privacy/scrubber.go

```diff
@@ -147,7 +147,10 @@ func (scrubber) ScrubRequest(bidRequest *openrtb2.BidRequest, enforcement Enforc
 	if enforcement.TID {
 		//remove source.tid and imp.ext.tid
 		if bidRequest.Source != nil {
-			bidRequest.Source.TID = ""
+			var sourceCopy *openrtb2.Source
+			sourceCopy = ptrutil.Clone(bidRequest.Source)
+			sourceCopy.TID = ""
```

Comment:
There is an issue here. If `bidRequest.Source` is `nil`, then `Clone` will return `nil`.  So you need to check for `nil` before setting `TID`.

Reply:
Here is the full "if case" portion of code related to the TID:
```go
if enforcement.TID {
		//remove source.tid and imp.ext.tid
		if bidRequest.Source != nil {
			var sourceCopy *openrtb2.Source
			sourceCopy = ptrutil.Clone(bidRequest.Source)
			sourceCopy.TID = ""
			bidRequest.Source = sourceCopy
		}
		for ind, imp := range bidRequest.Imp {
			impExt := scrubExtIDs(imp.Ext, "tid")
			bidRequest.Imp[ind].Ext = impExt
		}
	}
```
We will only clone source `if bidRequest.Source != nil`

Is this what you mean?

Reply:
I think @hhhjort missed the nil check. This looks fine to me.